internal/kibana/maintenance_window.go

```diff
@@ -0,0 +1,455 @@
+package kibana
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/kibana"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func validateMinMaintenanceWindowServerVersion(serverVersion *version.Version) diag.Diagnostics {
+	var maintenanceWindowPublicAPIMinSupportedVersion = version.Must(version.NewVersion("8.1.0"))
+	var diags diag.Diagnostics
+
+	if serverVersion.LessThan(maintenanceWindowPublicAPIMinSupportedVersion) {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Maintenance window API not supported",
+			Detail:   fmt.Sprintf(`The maintenance Window public API feature requires a minimum Elasticsearch version of "%s"`, maintenanceWindowPublicAPIMinSupportedVersion),
+		})
+		return diags
+	}
+	return nil
+}
+
+func ResourceMaintenanceWindow() *schema.Resource {
```

Comment:
Can we rebuild this as a plugin-framework based resource ([data view](https://github.com/elastic/terraform-provider-elasticstack/tree/main/internal/kibana/data_view) as an example). It automatically solves a bunch of TF related limitations, and should be the future for any new work in the provider.